### PR TITLE
Remove old thread API

### PR DESCRIPTION
**Pull request by mattcaswell**
This pull request completes the removal of the old threading API. No more locking callbacks!!

### CODE REVIEW

**Comment by levitte** [comment]
I'd like to see a line in NEWS as well

**Comment by levitte** [comment]
+1 when you've done so

**Comment by ghedo** [comment]
\o/

**Comment by levitte** [comment]
Funny, whenever I see `\o/`, I wanna shout "Doggy!"

([freefall](http://freefall.purrsia.com/) made me do it ;-))

**Comment by mattcaswell** [comment]
New commit added with entry in NEWS. @levitte still +1?

**Comment by levitte** [comment]
Yes

**Comment by mattcaswell** [comment]
Annnnnd merged!!!!!!!!!!!!!!!!

Thanks @ghedo, @levitte, @richsalz! It was a bit tight but we got it in by the deadline!

File: crypto/Makefile.in [hunk 1]
```diff
@@ -34,7 +34,7 @@ GENERAL=Makefile README crypto-lib.com install.com
 LIB= $(TOP)/libcrypto.a
 SHARED_LIB= libcrypto$(SHLIB_EXT)
 LIBSRC=	cryptlib.c mem.c mem_clr.c mem_dbg.c cversion.c ex_data.c cpt_err.c \
-	ebcdic.c uid.c o_time.c o_str.c o_dir.c thr_id.c lock.c \
+	ebcdic.c uid.c o_time.c o_str.c o_dir.c \
```
Comment (ghedo) [inline crypto/Makefile.in]: Oh, I think you need to updated `LIBOBJ` as well to fix classic builds
https://travis-ci.org/openssl/openssl/jobs/114791057#L5991

**Pull request closed without merging**